bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeDeviceHandler.java

```diff
@@ -0,0 +1,501 @@
+        if (deviceInfo == null) {
+            return null;
+        }
+
+        return deviceInfo;
+    }
+
+    protected void queryUsage() {
+        // Try to go online if the device was previously taken offline due to connection issues w/ cloud
+        if (getThing().getStatus() == ThingStatus.OFFLINE) {
+            goOnline();
+            return;
+        }
+
+        try {
+            tryQueryUsage(false);
+            tryGetCurrentFlowRate(false);
+        } catch (FlumeApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            this.handleApiException(e);
+            return;
+        }
+
+        if (System.nanoTime() > this.expiryUsageAlertFetch) {
+            fetchUsageAlerts();
+            this.expiryUsageAlertFetch = System.nanoTime() + USAGE_QUERY_FETCH_INTERVAL.toNanos();
+        }
+    }
+
+    public void fetchUsageAlerts() {
+        List<FlumeApiUsageAlert> resultList;
+        FlumeApiUsageAlert alert;
+        FlumeApiQueryWaterUsage query;
+
+        boolean imperialUnits = isImperial();
+
+        try {
+            resultList = getApi().fetchUsageAlerts(config.id, 1);
+        } catch (FlumeApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            this.handleApiException(e);
+            return;
+        }
+
+        if (resultList.isEmpty()) {
+            return;
+        }
+
+        alert = resultList.get(0);
+        // alert has already been notified or occurred before the device went online
+        if (!alert.triggeredDateTime.isAfter(this.lastUsageAlert)) {
+            logger.trace("alert: {}, lastUsageAlert: {}", alert.triggeredDateTime, this.lastUsageAlert);
+            return;
+        }
+
+        lastUsageAlert = alert.triggeredDateTime;
+
+        String stringAlertFormat = Objects.requireNonNull(getBridgeHandler())
+                .getLocaleString("trigger.high-flow-alert");
+        if (stringAlertFormat == null) {
+            return;
+        }
+
+        query = alert.query;
+        query.bucket = FlumeApi.BucketType.MIN;
+        query.operation = FlumeApi.OperationType.AVG;
+        query.units = imperialUnits ? FlumeApi.API_UNITS_IMPERIAL : FlumeApi.API_UNITS_METRIC;
+
+        Float avgUsage;
+        try {
+            avgUsage = getApi().queryUsage(config.id, query);
+        } catch (FlumeApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            this.handleApiException(e);
+            return;
+        }
+        long minutes = Duration.between(query.sinceDateTime, query.untilDateTime).toMinutes();
+
+        LocalDateTime whenTriggered = alert.triggeredDateTime.withZoneSameInstant(ZoneId.systemDefault())
+                .toLocalDateTime();
+
+        String stringAlert = String.format(stringAlertFormat, alert.eventRuleName, whenTriggered.toString(), minutes,
+                avgUsage, imperialUnits ? "gallons" : "liters");
+
```

Comment:
Similar comment as before about the recalculation between gallons and litre.

Reply:
this one is correct as one of the parameters to the query is units which is set several lines above.

Reply:
Yes, i see thanks.